cmd/kubeadm/app/phases/addons/dns/manifests.go

```diff
@@ -212,4 +212,169 @@ spec:
   selector:
     k8s-app: kube-dns
 `
+
+	// CoreDNSDeployment is the CoreDNS Deployment manifest
+	CoreDNSDeployment = `
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: coredns
+  namespace: kube-system
+  labels:
+    k8s-app: coredns
+    kubernetes.io/cluster-service: "true"
+    kubernetes.io/name: "CoreDNS"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: coredns
+  template:
+    metadata:
+      labels:
+        k8s-app: coredns
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+        scheduler.alpha.kubernetes.io/tolerations: '[{"key":"CriticalAddonsOnly", "operator":"Exists"}]'
+    spec:
+      serviceAccountName: coredns
+      tolerations:
+      - key: {{ .MasterTaintKey }}
+        effect: NoSchedule
+      containers:
+      - name: coredns
+        image: coredns/coredns:{{ .Version }}
+        imagePullPolicy: IfNotPresent
+        resources:
+          limits:
+            memory: 170Mi
+          requests:
+            cpu: 100m
+            memory: 70Mi
+        args: [ "-conf", "/etc/coredns/Corefile" ]
+        volumeMounts:
+        - name: config-volume
+          mountPath: /etc/coredns
+        ports:
+        - containerPort: 53
+          name: dns
+          protocol: UDP
+        - containerPort: 53
+          name: dns-tcp
+          protocol: TCP
+        - containerPort: 9153
+          name: metrics
+          protocol: TCP
+        livenessProbe:
+          httpGet:
+            path: /health
+            port: 8080
+            scheme: HTTP
+          initialDelaySeconds: 60
+          timeoutSeconds: 5
+          successThreshold: 1
+          failureThreshold: 5
+      dnsPolicy: Default
+      volumes:
+        - name: config-volume
+          configMap:
+            name: coredns
+            items:
+            - key: Corefile
+              path: Corefile
+`
+
+	// CoreDNSService is the CoreDNS Service manifest
+	CoreDNSService = `
+apiVersion: v1
+kind: Service
+metadata:
+  name: coredns
+  namespace: kube-system
+  labels:
+    k8s-app: coredns
+    kubernetes.io/cluster-service: "true"
+    kubernetes.io/name: "CoreDNS"
+spec:
+  selector:
+    k8s-app: coredns
+  clusterIP: {{ .DNSIP }}
+  ports:
+  - name: dns
+    port: 53
+    protocol: UDP
+  - name: dns-tcp
+    port: 53
+    protocol: TCP
+  - name: metrics
+    port: 9153
+    protocol: TCP
+`
+
+	//ConfigMap is the CoreDNS ConfigMap manifest
+	CoreDNSConfigMap = `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns
+  namespace: kube-system
+data:
+  Corefile: |
+    .:53 {
+        errors
+        log stdout
+        health
+        kubernetes {{ .DNSDomain }} {{ .ServiceCIDR }}
+        prometheus
+        proxy . /etc/resolv.conf
+        cache 30
+    }
+`
+	// CoreDNSClusterRole is the CoreDNS ClusterRole manifest
+	CoreDNSClusterRole = `
+apiVersion: rbac/v1
```

Comment:
group name is `rbac.authorization.k8s.io/v1`

Reply:
tagging @kubernetes/sig-auth-feature-requests and @kubernetes/sig-network-feature-requests -- how do you feel about making this an auto-bootstrapped role?

Reply:
in hindsight, the `kube-dns` role should have been in a manifest along with the add-on (serviceaccount + role definition and binding + deployment).

Reply:
So you mean that the kube-dns clusterrole _should not_ be auto-bootstrapped?
I disagree. I think it should be created by default.

I also think there is lots of value in having the clusterrolebinding created by default, but feel less strongly about that.

Reply:
> So you mean that the kube-dns clusterrole should not be auto-bootstrapped?
> I disagree. I think it should be created by default.

Why? The role is specific to kube-dns. 

> I also think there is lots of value in having the clusterrolebinding created by default

Again, why? That limits where and how you deploy the DNS pods. I'd like to see add-ons split into their own namespaces, with their own service accounts.

Reply:
@liggitt aren't critical addon pods restricted to the kube-system namespace currently?

Reply:
> @liggitt aren't critical addon pods restricted to the kube-system namespace currently?

Yes, but only because the current method of indicating them is an annotation. Long-term, pod priority would be used instead, and wouldn't treat `kube-system` specially